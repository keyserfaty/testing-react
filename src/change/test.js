import React from "react";
import { create } from "react-test-renderer";
import Change from ".";

describe('tests change', () => {
  const component = create(<Change />);
  const rootInstance = component.root;

  const input = rootInstance.findByType("input");
  const container = rootInstance.findByType("div");

  it("changes the value when typing on the input", () => {
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    
    input.props.onChange({ target: { name: 'item', value: 'some' } });
    
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    
    expect(container.props.children).toContain("some");
  });
})