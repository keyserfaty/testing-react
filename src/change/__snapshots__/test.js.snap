// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests change changes the value when typing on the input 1`] = `
<div>
  <input
    name="item"
    onChange={[Function]}
    value=""
  />
  Value: 
  
</div>
`;

exports[`tests change changes the value when typing on the input 2`] = `
<div>
  <input
    name="item"
    onChange={[Function]}
    value="some"
  />
  Value: 
  some
</div>
`;
